p2p/peer.py

```diff
@@ -342,7 +343,81 @@ def is_closing(self) -> bool:
     async def _cleanup(self) -> None:
         self.close()
 
+    async def boot(self) -> None:
+        try:
+            await self._boot()
+        except OperationCancelled:
+            # If a cancellation happens during boot we suppress it here and
+            # simply exit without setting the `booted` event.
+            return
+        else:
+            self.booted.set()
+
+    async def _boot(self) -> None:
+        try:
+            await self.ensure_same_side_on_dao_fork()
```

Comment:
How about we would do something like:

```python

if asyncio.isCoroutineFunction(self.boot_check):
    try:
       await self.boot_check()
    except OperationCancelled:
...
```
And then either pass the `boot_check` on creation or use an inheritence approach to set it.
That way we would keep this concrete DAO check out of the `BasePeer` which strikes me as being more generic in respect to other networks.

Reply:
This is another half-step towards the full solution.  Immediately following this PR is #1270 which completely moves this and other EVM stuff out of this module and into the `trinity` codebase.